src/components/controllers/repl-page.jsx

```diff
@@ -38,42 +35,31 @@ export default function ReplPage() {
  *
  * ?code -> ?example -> localStorage -> simple counter example
  */
-function initialCode(query) {
-	let code, slug;
+async function getInitialCode(query) {
+	const { route } = useLocation();
```

Comment:
Admittedly this block was a bit of a mess that I should have given a better look over.

Fixes an issue with `?code` being cleared on mount (though it would populate correctly) and drops `history.replaceState` in a few places in favor of `route()`.

This still isn't great, `preact-iso` has a rather fundamental issue at the moment (IMO) in that it updates the location context as soon as it matches a route. This leads to really awful control flow w/ suspense as you cannot rely on the content being updated alongside a path change -- content will update in a subsequent render. It's workable, just... fiddly. I was trying to avoid some of the woes by using `history.replaceState` but this refactor should work well while getting us in a position to take advantage of a fix to `iso` in the future (though I think it'll need a major).